src/OrbitDB.js

```diff
@@ -58,20 +61,39 @@ let databaseTypes = {
       throw new Error('IPFS is a required argument. See https://github.com/orbitdb/orbit-db/blob/master/API.md#createinstance')
 
     const { id } = await ipfs.id()
-    const directory = options.directory || './orbitdb'
-    const keystore = options.keystore || Keystore.create(path.join(directory, id, '/keystore'))
 
-    const identity = options.identity || await Identities.createIdentity({
-      id: options.id || id,
-      keystore: keystore,
-    })
-    options = Object.assign({}, options, {
-      peerId: id ,
-      directory: directory,
-      keystore: keystore
-    })
-    const orbitdb = new OrbitDB(ipfs, identity, options)
-    return orbitdb
+    if(!options.directory)
+      options.directory = './orbitdb'
+
+    if (!options.storage) {
+      const leveldown = isNode ? require('leveldown') : require('level-js')
+      options.storage = Storage(leveldown)
+      options.storage.preCreate = isNode ? async (directory, options) => {
```

Comment:
Since [Storage](https://github.com/orbitdb/orbit-db-storage-adapter/blob/master/src/index.js#L28) takes in `options`, how about passing the `preCreate` function as an option instead of assigning it here? I feel that would be a bit cleaner and doesn't require overriding the function (and we can remove https://github.com/orbitdb/orbit-db-storage-adapter/blob/master/src/index.js#L65).